lib/kitchen/element_enumerator_base.rb

```diff
@@ -101,6 +101,20 @@ def non_introduction_pages(only: nil, except: nil)
                except: except)
     end
 
+    # Returns an enumerator that iterates through both pages and composite pages, returning all as PageElements
+    #
+    # @param css_or_xpath [String] additional selectors to further narrow the element iterated over;
+    #   a "$" in this argument will be replaced with the default selector for the element being
+    #   iterated over.
+    #
+    def pages_and_composite_pages(only: nil, except: nil)
+      block_error_if(block_given?)
+      chain_to(PageElementEnumerator,
+               css_or_xpath: "div[data-type='page'], div[data-type='composite-page']",
+               only: only,
+               except: except)
+    end
+
```

Comment:
Is there a way to dynamically cast elements as either PageElements or CompositePageElements?

Reply:
Yes, there is a `TypeCastingElementEnumerator`.  Also `ElementBase#search_with`, which maybe is unused at the moment.  I'd look at these, might need to add `except` and `only` to `search_with`